### DISCUSSION TITLE
Best practice for mapping visit_detail_concept_id value

### DISCUSSION THREAD

**Discussion by mkrile38** [chorus-ai/StandardsModule#94] [Structured EHR and ETL Questions] [answered]
I am wondering if the cited 'Visit' domain as specified in OMOP CDM v5.4 visit_detail section is the best set for mapping visit_detail_concept_id. For our visits, we found more suitable visit descriptors in that fell in the Observation domain.

I attached our mapping file for an example. The source concepts are the ones we mapped to internally, and the target fields are our attempts to map to visit domain concepts.

Any guidance is greatly appreciated.

[care_site_concepts_PITT.xlsx](https://github.com/user-attachments/files/16129565/care_site_concepts_PITT.xlsx)

**Comment by heschmidt04** [comment]
Hope this helps folks. 

At MGB we do have visit types that fall outside of what is matched up with the standard concepts in Concept Vocab table with          
```
SELECT
            co.concept_id  AS target_concept_id,
        FROM
             concept co 
        WHERE LOWER(co.domain_id) = 'visit'
        WHERE LOWER(co.standard_concept) = 's' ;
```

For the data set we have these handle the majority of the visit types, we have about 5 visit types currently unaccounted for so they will fail DQD at the moment. 
Those are not as frequent in our current data set so they are on the back burner to solve at the moment. 

```
ADMITTING_SOURCE_CONCEPT_ID = '4144684'     WHERE ADMITTING_SOURCE_VALUE ='Self Referral';
ADMITTING_SOURCE_CONCEPT_ID = '40315731'    WHERE ADMITTING_SOURCE_VALUE ='Psych, Substance Abuse, or Rehab Hospital';
DISCHARGED_TO_CONCEPT_ID = '8536'  			WHERE DISCHARGED_TO_SOURCE_VALUE ='Home or Self Care';
DISCHARGED_TO_CONCEPT_ID = '8536'  			WHERE DISCHARGED_TO_SOURCE_VALUE ='Home-Health Care Svc';
DISCHARGED_TO_CONCEPT_ID = '32217' 			WHERE DISCHARGED_TO_SOURCE_VALUE ='Admitted as an Inpatient';
DISCHARGED_TO_CONCEPT_ID = '32234' 			WHERE DISCHARGED_TO_SOURCE_VALUE ='Another Health Care Institution Not Defined';
DISCHARGED_TO_CONCEPT_ID = '8970'  			WHERE DISCHARGED_TO_SOURCE_VALUE ='Long Term Care';
DISCHARGED_TO_CONCEPT_ID = '32218' 			WHERE DISCHARGED_TO_SOURCE_VALUE ='Expired';
DISCHARGED_TO_CONCEPT_ID = '32211' 			WHERE DISCHARGED_TO_SOURCE_VALUE ='Short Term Hospital';
DISCHARGED_TO_CONCEPT_ID = '32216' 			WHERE DISCHARGED_TO_SOURCE_VALUE ='Left Against Medical Advice';
DISCHARGED_TO_CONCEPT_ID =  '8970' 			WHERE DISCHARGED_TO_SOURCE_VALUE ='Federal Hospital';
DISCHARGED_TO_CONCEPT_ID = '32220' 			WHERE DISCHARGED_TO_SOURCE_VALUE ='Still a Patient (DC Readmit Only)';

```

**Comment by jshoughtaling** [comment, marked as answer]
Thanks, @heschmidt04, for following up on this question! 

@mkrile38 - thanks for the question! The `visit_detail_concept_id` field is reserved for concepts in the VISIT domain only, and we advise you to _not_ use concepts from other domains in this field. Many different OMOP-related tools rely on proper domain association of concepts to work properly, and VISIT_DETAIL is no exception. If you have source descriptions without any reasonable match in the VISIT domain, it's possible that we can create CHoRUS specific concepts for VISIT_DETAIL events that match your descriptions. Let's plan to discuss further offline

**Comment by heschmidt04** [comment]
This is more of a side bar, but ties into visit_detail.

I was seeing a pattern in our ADT data that showed a Discharge event after a "Hospital Outpatient" event that was for a procedure (MRI, XR, CUPID, CT Head, Lab long term monitoring, COX2, etc.) and noticed that our logic didn't catch that combination.
I chose 38004287 = 'Procedural Series' for that pattern because it was in our source_to_concept_map as a visit admit concept code, is standard, and reflects that a procedure was done. 
I added to our source to concept map as a visit discharge concept code. 

Also, for MGB data I made the decision to use code 32216 for anything that indicated leaving or not showing up; including "Elope to another provider" (which tickled my funny bone) 

Hope that ends up helping folks who get to this point of tracking down visit types (admit and discharge) that DQD (Data Quality Dashboard) determines have empty values.
